src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfoOfTConverter.cs

```diff
@@ -0,0 +1,188 @@
+
+        public override JsonConverter ConverterBase
+        {
+            get
+            {
+                return Converter;
+            }
+            set
+            {
+                Debug.Assert(value is JsonConverter<TConverter>);
+                Converter = (JsonConverter<TConverter>)value;
+            }
+        }
+
+        public override object? GetValueAsObject(object obj)
+        {
+            if (IsPropertyPolicy)
+            {
+                return obj;
+            }
+
+            Debug.Assert(HasGetter);
+            return Get!(obj);
+        }
+
+        public override bool GetMemberAndWriteJson(object obj, ref WriteStack state, Utf8JsonWriter writer)
+        {
+            Debug.Assert(EscapedName.HasValue);
+
+            bool success;
+            TConverter value = Get!(obj);
+            if (value == null)
+            {
+                if (!IgnoreNullValues)
+                {
+                    writer.WriteNull(EscapedName.Value);
+                }
+
+                success = true;
+            }
+            else
+            {
+                if (state.Current.PropertyState < StackFramePropertyState.Name)
+                {
+                    state.Current.PropertyState = StackFramePropertyState.Name;
+                    writer.WritePropertyName(EscapedName.Value);
+                }
+
+                success = Converter.TryWrite(writer, value, Options, ref state);
+            }
+
+            return success;
+        }
+
+        public override bool GetMemberAndWriteJsonExtensionData(object obj, ref WriteStack state, Utf8JsonWriter writer)
+        {
+            bool success;
+            TConverter value = Get!(obj);
+
+            if (value == null)
+            {
+                success = true;
+            }
+            else
+            {
+                state.Current.PolymorphicJsonPropertyInfo = state.Current.DeclaredJsonPropertyInfo!.RuntimeClassInfo.ElementClassInfo!.PolicyProperty;
+                success = Converter.TryWriteDataExtensionProperty(writer, value, Options, ref state);
+            }
+
+            return success;
+        }
+
+        public override bool ReadJsonAndSetMember(object obj, ref ReadStack state, ref Utf8JsonReader reader)
+        {
+            bool success;
+            bool isNullToken = reader.TokenType == JsonTokenType.Null;
+            if (isNullToken && !Converter.HandleNullValue && !state.IsContinuation)
+            {
+                if (!IgnoreNullValues)
+                {
+                    TConverter value = default!;
+                    Set!(obj, value);
+                }
+
+                success = true;
+            }
+            else
+            {
+                // Optimize for internal converters by avoiding the extra call to TryRead.
+                if (Converter.CanUseDirectReadOrWrite)
+                {
+                    TConverter fastvalue = Converter.Read(ref reader, RuntimePropertyType!, Options);
```

Comment:
We are potentially passing in `reader.TokenType == JsonTokenType.Null` here, when we probably shouldn't. No?

Reply:
Yes that is why this code path is only for internal converters.